src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -0,0 +1,22 @@
+import { type Token } from '@dfinity/utils';
+
+/**
+ * Calculates the maximum amount for a transaction.
+ *
+ * @param {Object} params
+ * @param {bigint | undefined} params.balance The balance of the account.
+ * @param {bigint | undefined} params.fee The fee of the transaction.
+ * @returns {number} The maximum amount for the transaction.
+ */
+export const getMaxTransactionAmount = ({
```

Comment:
You are correct!

I don't know if it is overcomplicated, but I would pass the feeSymbol too then, and if it's different, it will not subtract.

```typescript
export const getMaxTransactionAmount = ({
  balance = 0n,
  fee = 0n,
  token,
  feeToken
}: {
  balance?: bigint;
  fee?: bigint;
  token: Token;
  feeToken?: Token | undefined;
}): number => {
  if (feeToken !== undefined && feeToken.symbol !== token.symbol) {
    return Math.max(Number(balance), 0) / 10 ** token.decimals;
  }
  return Math.max(Number(balance - fee), 0) / 10 ** token.decimals;
};
```

Too much?